Specs/addDefaultMatchers.js

```diff
@@ -216,6 +417,200 @@ define([
         };
     }
 
+    function renderAndReadPixels(options) {
+        var scene;
+
+        if (defined(options.scene)) {
+            // options were passed to render the scene at a given time or prime shadow map
+            scene = options.scene;
+            var time = options.time;
+
+            scene.initializeFrame();
+            if (defined(options.primeShadowMap)) {
+                scene.render(time); // Computes shadow near/far for next frame
+            }
+            scene.render(time);
+        } else {
+            scene = options;
+            scene.initializeFrame();
+            scene.render();
+        }
+
+        return scene.context.readPixels();
+    }
+
+    function renderEquals(util, customEqualityTesters, actual, expected, expectEqual) {
+        var actualRgba = renderAndReadPixels(actual);
+
+        // When the WebGL stub is used, all WebGL function calls are noops so
+        // the expectation is not verified.  This allows running all the WebGL
+        // tests, to exercise as much Cesium code as possible, even if the system
+        // doesn't have a WebGL implementation or a reliable one.
+        if (!!window.webglStub) {
+            return {
+                pass : true
+            };
+        }
+
+        var eq = equals(util, customEqualityTesters, actualRgba, expected);
+        var pass = expectEqual ? eq : !eq;
+
+        var message;
+        if (!pass) {
+            message = 'Expected ' + (expectEqual ? '' : 'not ')  + 'to render [' + expected + '], but actually rendered [' + actualRgba + '].';
+        }
+
+        return {
+            pass : pass,
+            message : message
+        };
+    }
+
+    function pickPrimitiveEquals(actual, expected) {
+        var scene = actual;
+        var result = scene.pick(new Cartesian2(0, 0));
+
+        var pass = true;
+        var message;
+
+        var webglStub = !!window.webglStub;
+        if (!webglStub) {
+            if (defined(expected)) {
+                pass = (result.primitive === expected);
+            } else {
+                pass = !defined(result);
+            }
+
+            if (!pass) {
+                message = 'Expected to pick ' + expected + ', but picked: ' + result;
+            }
+        }
```

Comment:
Is it possible for this one to follow the organization of `renderEquals` more closely (like return early)?